### ISSUE TITLE
PngWriter did not write meta data into output stream.

### ISSUE THREAD

**Issue by leonbloy** [leonbloy/pngj#29]
_From [own...@t8ke.com](https://code.google.com/u/102791969586225107214/) on March 24, 2014 01:29:29_

What steps will reproduce the problem? 1. Create a Png Image using PngWriter like the following (psuedo code)
		int side = 80;
		ByteArrayOutputStream os = new ByteArrayOutputStream();
		ImageInfo imageInfo = new ImageInfo(side, side, 8, true); // 8 bits true color RGBA
		PngWriter png = new PngWriter(os, imageInfo);
                
               png.getMetadata().setDpi(300);
               // ... and more setting ..
         
              //write the png
              png.end();

2. The image looks good. However, the meta data is not shown. DPI is set to 72 instead of 300.
3. A quick scan of the code indicate that meta data object is not written to file. What is the expected output? What do you see instead? Write the meta data to the PNG file - for me, the DPI setting is very important (for printing purpose). What version of the product are you using? On what operating system? 2.0.1. on Java / MacOS (testing) / AppEngine (production) Please provide any additional information below.

_Original issue: http://code.google.com/p/pngj/issues/detail?id=29_

**Comment by leonbloy** [comment]
_From [ma.qian...@gmail.com](https://code.google.com/u/106325105330383636495/) on March 25, 2014 22:06:02_

It seems I could not edit this issue. I think I know why now. There is an issue when I used copy chunk from background. I will create a test case and redo the issue description.

**Comment by leonbloy** [comment]
_From [ma.qian...@gmail.com](https://code.google.com/u/106325105330383636495/) on March 25, 2014 22:32:17_

Ah. Found the problem. It was my coding issue. Resolved. Sorry about this!

**Comment by leonbloy** [comment]
_From [hjg.com.ar@gmail.com](https://code.google.com/u/hjg.com.ar@gmail.com/) on March 26, 2014 03:56:49_

No problem. In case the issue was not obvious (and it might help others, or make me retouch the implementation or the API) perhaps you'd want to copy the malfunctioning code in more detail?

**Comment by leonbloy** [comment]
_From [ma.qian...@gmail.com](https://code.google.com/u/106325105330383636495/) on March 26, 2014 08:44:06_

Let me explain what I did wrong. What confused me was the "chunk" vs "metadata". Looking into a bit more detail, it appears that "chunk" is really an implementation of "metadata". In my case, I need to copy a background picture and then draw on the background picture.  what I did -
     create new PngWriter; 
     create new PngReader from the background image;
     set meta data on PngWriter;
     pngWriter.copyChunksFrom(backgroundImage.getChunksList(), ChunkCopyBehaviour.COPY_ALL_SAFE); // error line

Basically, in the background, set meta data set the meta data chunk. Then copy did it again.
Interestingly, that messed the meta data chunk up. The DPI in both setter (meta data and copy) are 300. After doing both, I got 72. 

Anyway, after removing the last line, it worked perfectly.

Thank you so much for the library. It really helps!
